include/seqan/align/dp_formula.h

```diff
@@ -110,101 +114,104 @@ typedef Tag<RecursionDirectionZero_> RecursionDirectionZero;
 // Function _computeScore
 // ----------------------------------------------------------------------------
 
-template <typename TScoreValue, typename TGapCosts, typename TSequenceHValue, typename TSequenceVValue,
-          typename TScoringScheme, typename TRecursionDirection, typename TDPProfile>
-inline SEQAN_FUNC_ENABLE_IF(Not<Is<SimdVectorConcept<TScoreValue> > >, typename TraceBitMap_<TScoreValue>::Type)
-_computeScore(DPCell_<TScoreValue, TGapCosts> & activeCell,
-              DPCell_<TScoreValue, TGapCosts> const & previousDiagonal,
-              DPCell_<TScoreValue, TGapCosts> const & previousHorizontal,
-              DPCell_<TScoreValue, TGapCosts> const & previousVertical,
+template <typename TScoreNavigator,
+          typename TSequenceHValue,
+          typename TSequenceVValue,
+          typename TScoringScheme,
+          typename TRecursionDirection,
+          typename TDPProfile>
+inline SEQAN_FUNC_ENABLE_IF(Not<Is<SimdVectorConcept<ExtractedScoreValueType_<TScoreNavigator>>>>,
+                            typename TraceBitMap_<ExtractedScoreValueType_<TScoreNavigator>>::Type)
+_computeScore(TScoreNavigator && dpNavi,
               TSequenceHValue const & seqHVal,
               TSequenceVValue const & seqVVal,
               TScoringScheme const & scoringScheme,
               TRecursionDirection const & recDir,
               TDPProfile const & dpProfile)
 {
-    typedef typename TraceBitMap_<TScoreValue>::Type TTraceValue;
-
-    TTraceValue traceDir = _doComputeScore(activeCell, previousDiagonal, previousHorizontal, previousVertical, seqHVal,
-                                           seqVVal, scoringScheme, recDir, dpProfile);
+    auto traceDir = _doComputeScore(std::get<0>(dpNavi), std::get<1>(dpNavi), std::get<2>(dpNavi),
+                                    std::get<3>(dpNavi), seqHVal, seqVVal, scoringScheme, recDir, dpProfile);
     if (IsLocalAlignment_<TDPProfile>::VALUE)
-        if (activeCell._score <= 0)
+    {
+        if (_scoreOfCell(std::get<0>(dpNavi)) <= 0)
         {
-            _setScoreOfCell(activeCell, static_cast<TScoreValue>(0));
-            _setHorizontalScoreOfCell(activeCell, static_cast<TScoreValue>(0));
-            _setVerticalScoreOfCell(activeCell, static_cast<TScoreValue>(0));
-            return TraceBitMap_<TScoreValue>::NONE;
+            _setScoreOfCell(std::get<0>(dpNavi), static_cast<ExtractedScoreValueType_<TScoreNavigator>>(0));
```

Comment:
There are different internalComputeScore interfaces for different Gap specialisations. 
So I kept it generic.